### PR TITLE
fix: compatibility with routing-controllers 0.9

### PR DESCRIPTION
**Pull request by mgcostaParedes**
this PR will fix the compatibility with routing-controllers **0.9**.

class-transformer has been also updated, since the defaultMetadaStorage has been migrated to a new location, had to do a little workout. more info on: https://github.com/typestack/class-transformer/issues/563.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -16,10 +16,11 @@
     "strictNullChecks": true,
     "forceConsistentCasingInFileNames": true,
     "noFallthroughCasesInSwitch": true,
-    "noImplicitAny": true,
     "noImplicitThis": true,
     "noUnusedLocals": true,
-    "noUnusedParameters": true
+    "noUnusedParameters": true,
+    "allowSyntheticDefaultImports": true,
+    "allowJs": true
```
Comment (epiphone) [inline tsconfig.json:23]: Why are these changes needed?

File: .travis.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ language: node_js
 node_js:
   - 12
 script:
+  - yarn format
```
Comment (epiphone) [inline .travis.yml:6]: Please revert this. Formatting code before `test:format` defeats the whole purpose of the format check :)

File: package.json [hunk 3]
```diff
@@ -43,9 +43,9 @@
     "@types/prettier": "^2.0.0",
     "@types/reflect-metadata": "^0.1.0",
     "@types/rimraf": "^3.0.0",
-    "class-transformer": "^0.3.1",
+    "class-transformer": "^0.4.0",
     "class-validator": "^0.12.2",
-    "class-validator-jsonschema": "^2.0.0-rc1",
+    "class-validator-jsonschema": "^2.2.0",
```
Comment (epiphone) [inline package.json:48]: Note that `class-validator-jsonschema` isn't compatible with `class-transformer >0.3.1` atm: https://github.com/epiphone/class-validator-jsonschema/issues/55

**Review by epiphone — CHANGES_REQUESTED** [review]
Thanks! I'll try to find time to take a closer look later

File: .travis.yml (hunk 2 above)
Comment (mgcostaParedes) [reply .travis.yml]: done, reverted ;)

File: package.json (hunk 3 above)
Comment (mgcostaParedes) [reply package.json]: reverted to the 2.0.0-rc1

File: tsconfig.json (hunk 1 above)
Comment (mgcostaParedes) [reply tsconfig.json]: we just need to remove the **noImplicityAny** to be available to import the defaultMetadataStorage from the new version of class-transformer, otherwise would dispatch an error, the last 2 lines, are not necessary, reverted.

**Comment by mgcostaParedes** [comment]
> Thanks! I'll try to find time to take a closer look later

can you check @epiphone if is that ok now?

File: tsconfig.json (hunk 1 above)
Comment (epiphone) [reply tsconfig.json]: Is there any way to use an explicit `as any` coercion and keep `"noImplicitAny": true`?

**Comment by epiphone** [comment]
I just upgraded a bunch of dependencies in `master` which causes a bunch of conflicts here; I'll open another PR and cherry-pick you're commit there. Thanks!
